### PR TITLE
chore: added global files to contain test selector mappers and public data

### LINKED ISSUE #1126: test: add global files to contain selector mappers and public methods
# Feature request

## Please describe your feature

frontend E2E test:
add a global file to contain selector mappers 
add some public methods, such as `create base route`, `create base upstream` etc.

### PR DESCRIPTION
**Pull request by guoqqqi**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
resolve #1126 
___
### New feature or improvement
- Describe the details and related test reports.
Add Public Selector
___

### CODE REVIEW

**Comment by guoqqqi** [comment]
Public data has not been updated yet

File: web/cypress/fixtures/selector.json [hunk 1]
```diff
@@ -1,7 +1,49 @@
 {
+  "switcher": ".ant-space-align-center",
```
Comment (liuxiran) [inline web/cypress/fixtures/selector.json:2]: `switcher` -> `language siwtcher`

**Comment by liuxiran** [comment]
It would be better to keep the selectors only for individual cases in the case files

**Comment by juzhiyuan** [comment]
Hi @liuxiran, how about putting public(repeated) selectors/data to `fixtures` and others to every test case file?

**Comment by liuxiran** [comment]
> Hi @liuxiran, how about putting public(repeated) selectors/data to `fixtures` and others to every test case file?

agree 💯

File: web/cypress/fixtures/selector.json [hunk 2]
```diff
@@ -1,7 +1,49 @@
...
   "dropdown": ".rc-virtual-list",
   "notification": ".ant-notification-notice-message",
   "drawerBody": ".ant-drawer-wrapper-body",
   "notificationCloseIcon": ".ant-notification-close-icon",
-  "notificationDesc": ".ant-notification-notice-description"
+  "notificationDesc": ".ant-notification-notice-description",
+
+  "pluginsCard": ".ant-card",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:9]:
```suggestion
  "pluginCard": ".ant-card",
```

File: web/cypress/fixtures/selector.json [hunk 3]
```diff
@@ -1,7 +1,49 @@
...
+  "pageContent": ".ant-pro-page-container",
+
+  "tableCell": ".ant-table-cell",
+  "empty": ".ant-empty-normal",
+  "refresh": ".anticon-reload",
+  "codemirror": ".CodeMirror",
+  "switch": "#disable",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:16]: Do you mean `disabledSwitcher`?

File: web/cypress/fixtures/selector.json [hunk 4]
```diff
@@ -1,7 +1,49 @@
...
+  "deleteBtn": ".ant-btn-dangerous",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:17]:
```suggestion
  "deleteButton": ".ant-btn-dangerous",
```

File: web/cypress/fixtures/selector.json [hunk 5]
```diff
@@ -1,7 +1,49 @@
...
+
+  "name": "#name",
+  "nodes_0_host": "#nodes_0_host",
+  "upstream_id": "#upstream_id",
+  "input": ":input",
+  "custom_upstream": "[title=Custom]",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:23]: Do you mean `upstreamSelector?`

File: web/cypress/fixtures/selector.json [hunk 6]
```diff
@@ -1,7 +1,49 @@
...
+  "title_name": "[title=Name]",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:24]: Do you mean `nameSelector`?

File: web/cypress/fixtures/selector.json [hunk 7]
```diff
@@ -1,7 +1,49 @@
...
+  "test_service": "[title=test_service]",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:25]: Do you mean `serviceSelector`?

File: web/cypress/fixtures/selector.json [hunk 8]
```diff
@@ -1,7 +1,49 @@
...
+
+  "description": "#desc",
+  "search_name": "[title=Name]",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:28]: Do you mean `nameSelector`?

File: web/cypress/fixtures/selector.json [hunk 9]
```diff
@@ -1,7 +1,49 @@
...
+  "upstream_selector": "[data-cy=upstream_selector]",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:29]: Should be the same as other `upstreamSelector`

File: web/cypress/fixtures/selector.json [hunk 10]
```diff
@@ -1,7 +1,49 @@
...
+
+  "file": "[type=file]",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:31]: should be more detailed

File: web/cypress/fixtures/selector.json [hunk 11]
```diff
@@ -1,7 +1,49 @@
...
+  "fileTypeRadio": "[type=radio]",
+
+  "uriInput": "#debugUri",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:34]: should have a good name

File: web/cypress/fixtures/selector.json [hunk 12]
```diff
@@ -1,7 +1,49 @@
...
+
+  "hosts_0": "#hosts_0",
+  "labels_0_labelKey": "#labels_0_labelKey",
+  "labels_0_labelValue": "#labels_0_labelValue",
+  "searchName": "[title=Name]",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:39]: ditto

File: web/cypress/fixtures/selector.json [hunk 13]
```diff
@@ -1,7 +1,49 @@
...
+  "searchLabels": "[title=Labels]",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:40]: ditto

File: web/cypress/fixtures/selector.json [hunk 14]
```diff
@@ -1,7 +1,49 @@
...
+
+  "pageContainer": ".ant-pro-page-container",
+  "notificationMsg": ".ant-notification-notice-message",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:43]: `notificationMessage`

File: web/cypress/fixtures/selector.json [hunk 15]
```diff
@@ -1,7 +1,49 @@
...
+  "setting": ".ant-space-align-center",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:44]: Name & Value don't have a relation IMO

File: web/cypress/fixtures/selector.json [hunk 16]
```diff
@@ -1,7 +1,49 @@
...
+  "grafanaURL": "#grafanaURL",
+  "explain": ".ant-form-item-explain",
+
+  "selectItem": ".ant-select-item-option-content"
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:48]: have a better name?

File: web/cypress/integration/route/create-edit-delete-route.spec.js [hunk 17]
```diff
@@ -22,8 +22,9 @@ context('Create and Delete Route', () => {
   const sleepTime = 100;
 
   beforeEach(() => {
-    // init login
     cy.login();
+
+    cy.fixture('selector.json').as('domSelector');
```
Comment (juzhiyuan) [inline web/cypress/integration/route/create-edit-delete-route.spec.js:27]: Not done?

File: web/cypress/integration/metrics/metrics-smoketest.spec.js [hunk 18]
```diff
@@ -17,20 +17,17 @@
...
-
   beforeEach(() => {
     cy.login();
+
+    cy.fixture('selector.json').as('domSelector');
   });
 
-  it('visit metrics page', () => {
-    // go to metrics page
+  it('visit metrics page', function () {
```
Comment (LiteSun) [inline web/cypress/integration/metrics/metrics-smoketest.spec.js:26]:
```suggestion
  it('should visit metrics page', function () {
```

**Comment by LiteSun** [comment]
This PR looks good to me, but I have a question to confirm, if a selector is not public, should we extract a variable in that case? @guoqqqi

File: web/cypress/fixtures/selector.json [hunk 19]
```diff
@@ -1,49 +1,46 @@
...
-  "search_name": "[title=Name]",
-  "upstream_selector": "[data-cy=upstream_selector]",
+  "upstreamSelector": "[data-cy=upstream_selector]",
 
-  "file": "[type=file]",
+  "fileSelector": "[type=file]",
   "fileTypeRadio": "[type=radio]",
 
-  "uriInput": "#debugUri",
+  "debugUri": "#debugUri",
```
Comment (juzhiyuan) [inline web/cypress/fixtures/selector.json:32]: selector? input box? or something else?

File: web/cypress/integration/route/search-route.spec.js [hunk 20]
```diff
@@ -77,21 +77,21 @@ context('Create and Search Route', () => {
...
-    cy.get(this.domSelector.searchName).type(data.test);
+    cy.get(this.domSelector.nameSelector).type(data.test);
     cy.contains('Search').click();
     cy.contains(data.test0).siblings().should('contain', data.desc0);
     cy.contains(data.test1).siblings().should('contain', data.desc1);
     cy.contains(data.test2).siblings().should('contain', data.desc2);
     // no match
     cy.reload();
-    cy.get(this.domSelector.searchName).type(data.testx);
+    cy.get(this.domSelector.nameSelector).type(data.testx);
```
Comment (juzhiyuan) [inline web/cypress/integration/route/search-route.spec.js:94]: what's name selector? it seems like a `nameInput` here.

**Comment by juzhiyuan** [comment]
don't forget to sync codes from the master branch.

**Comment by guoqqqi** [comment]
> This PR looks good to me, but I have a question to confirm, if a selector is not public, should we extract a variable in that case? @guoqqqi

I think we'd better put all the variables in the public file, not only to keep the formatting consistent, but we may use any of them later, what do you think？

**Comment by juzhiyuan** [comment]
NOT: all reusable variables、selectors

File: web/cypress/fixtures/selector.json (hunk 19 above)
Comment (guoqqqi) [reply web/cypress/fixtures/selector.json:43]: image
...emm, or just ues `uriInput`?

**Comment by guoqqqi** [comment]
All right.

**Review by juzhiyuan — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Pull request merged**